### PR TITLE
docs: alias concept

### PR DESCRIPTION
**Pull request by samuelstroschein**
Adds the alias concept

### CODE REVIEW

**Comment by samuelstroschein** [comment]
@NilsJacobsen I need 3 things from you: 

1. An icon for the alias concept in the left nav bar
2. Illustration for loosing change history (I think we can communicate much of this paragraph as an illustration)
3. Illustration Avoiding naming discussions (I think we can communicate much of this paragraph as an illustration)

CleanShot 2024-02-07 at 21 33 29@2x

**Comment by samuelstroschein** [comment]
@jldec @NilsJacobsen ready for review. 

CleanShot 2024-02-08 at 10 27 48@2x

**Comment by NilsJacobsen** [comment]
Is there also a place where we say that we use ids? I guess a lot of people know keys and this is kinda new right?

**Comment by samuelstroschein** [comment]
> Is there also a place where we say that we use ids? I guess a lot of people know keys and this is kinda new right?

Should probably be in the message concept. A section with #ID. Let's merge this here first though

**Review by NilsJacobsen — APPROVED** [review]

**Comment by felixhaeberle** [comment]
why is message alias not a subchapter in message, but a dedicated site?

**Comment by samuelstroschein** [comment]
@felixhaeberle huh. this is a good remark!

I will put that under `message` as well as an explainer on human readable ids

File: inlang/documentation/ecosystem/concept/alias.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Message Alias
+
+**❗ Only use aliases if you import pre-existing messages, not for newly created messages.** 
+
+Aliases allow managing pre-existing messages with keys like "account-settings-button-label" with the inlang ecosystem by treating the key as an "alias" for a message.
```
Comment (jldec) [inline inlang/documentation/ecosystem/concept/alias.md:5]: how about:

> Aliases provide a way to reference messages using pre-existing message identifers. 
>
> This is useful when you import messages and your code already has references to those messages.
> If your project is configured to use a plugin for message storage, the ids from the plugin are treated as aliases.

File: inlang/documentation/ecosystem/concept/alias.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
++  ios: "LOGIN_CARD_HEADER"  
++}
+```
+
+_A message always has an auto-generated but human readable ID. The alias is an optional field._
+
+## When to use aliases? 
+
+1. You have pre-existing messages with keys and want to adopt parts of inlang ecosystem without refactoring code.
+2. You target multiple platforms that need a different syntax to reference the same message e.g. iOS must use `UPPERCASE_WITH_UNDERSCORES` and Android must use `kebap-case`.
```
Comment (jldec) [inline inlang/documentation/ecosystem/concept/alias.md:21]: s/kebap/kebab/

File: inlang/documentation/ecosystem/concept/alias.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+3. Translators and designers have no possibility to create messages via inlang apps like [fink](/m/tdozzpar) or [parrot](m/gkrpgoir).
+  
+GitHub issue [#1892](https://github.com/opral/monorepo/issues/1892) provides more context on our decision to use auto-generated but human readable IDs for messages instead of keys.
+
+
+### Unnecessary name convention discussions
+
+![naming](https://cdn.jsdelivr.net/gh/opral/monorepo@latest/inlang/documentation/ecosystem/assets/alias-naming.png)
+
+The nature of given a something a "name" (key/alias) leads to naming convention discussions.
```
Comment (jldec) [inline inlang/documentation/ecosystem/concept/alias.md:40]: How about
> Giving something a "name" invitably leads to naming convention discussions.

File: inlang/documentation/ecosystem/concept/alias.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+### Unnecessary name convention discussions
+
+![naming](https://cdn.jsdelivr.net/gh/opral/monorepo@latest/inlang/documentation/ecosystem/assets/alias-naming.png)
+
+The nature of given a something a "name" (key/alias) leads to naming convention discussions. 
+
+Should a key be `login-button-label`, `login_button_label`, or even `loginScreenButton`? In reality, it does not matter. The main purpose of a key is to reference the message. Context like "Where a message is used?" is provided with tooling from the inlang ecosystem. Any attempt to establish a naming convention will lead to discussions. Discussions that are unnecessary because someone will, by mistake, break the naming convention and thereby make all discussions irrelevant. 
+
+Don't waste time discussing naming conventions. Use the assigned human readable IDs and ship your software. The inlang ecosystem tooling will do the rest! :)
```
Comment (jldec) [inline inlang/documentation/ecosystem/concept/alias.md:44]: ❤️ this.

File: inlang/documentation/ecosystem/concept/alias.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Don't waste time discussing naming conventions. Use the assigned human readable IDs and ship your software. The inlang ecosystem tooling will do the rest! :)  
+
+
+### Losing the change history of messages
+
+![lackOfImmutability](https://cdn.jsdelivr.net/gh/opral/monorepo@latest/inlang/documentation/ecosystem/assets/alias-key-immutability.png)
+
+Keys/aliases like "login-button-label" are mutable. Anyone can rename a key/alias. The moment a key/alias is renamed, history like "What was the previous English text?" of a message is lost.
+
```
Comment (jldec) [inline inlang/documentation/ecosystem/concept/alias.md:52]: how about breaking the paragraph and starting the 2nd one with: 
> Some existing systems try to...

**Review by jldec — APPROVED** [review]
LGTM 👍 

Nice - this will help a lot - a few small suggestions and a typo-fix in the comments.

We will probably also need a page specifically aboout our new **message ids**.
Describe how we auto-generate human id's. Link to this page in the section about imported messages.

File: inlang/documentation/ecosystem/concept/alias.md (hunk 1 above)
Comment (samuelstroschein) [reply inlang/documentation/ecosystem/concept/alias.md:5]: id != key

pre-existing messages with keys have no identifier. that's the problem. i don't think we should mix id and key interchangeably

**Comment by samuelstroschein** [comment]
merging this. subsequent improvements can be done with follow up prs

**Comment by jldec** [comment]
❤️

**Pull request merged**